storey/targets.py

```diff
@@ -450,9 +460,12 @@ def _event_to_batch_entry(self, event):
 
     async def _emit(self, batch, batch_key, batch_time, last_event_time=None):
         df_columns = []
-        if self._index_cols:
-            df_columns.extend(self._index_cols)
-        df_columns.extend(self._non_partition_columns)
+        if self._non_partition_columns:
```

Comment:
is this related to the infer_columns_from_data change or additional unrelated change?

Reply:
It's related. Without this change, you may get a schema with only the index columns, which we do not want.